handbook/engineering/search/index.md

```diff
@@ -2,9 +2,99 @@
 
 We own the end-to-end Sourcegraph search experience.
 
+## Vision
+
+Sourcegraph search is:
+
+- **Fast:** Performance of showing results is fast and I can get to what I’m looking for quickly.
+- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write. We strive for a search experience where the answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
```

Comment:
Suggested addition::

**Scalable:** Sourcegraph scales to any private codebase and the size of the open-source universe. Users can trust that the result set is drawn from the entire set of code they care about, and if the result isn't found, it doesn't exist.

Reply:
I think this is a good/important addition and zero objection to adding it. The one thing I'd call out here though is that the scope of a solution potentially goes beyond search. If we're talking about fast indexed search, sure, we can think in terms of horizontally scaling Zoekt, say. If we're talking about "exhaustively search for all instances so that I can replace X with Y in a campaign" it demands a search/architecture solution and scope that goes a bit beyond what I think the search team (at least relative to its current size) can focus on. I could see this being part of a core-like team that potentially branches out of a search concern.

Reply:
It is ok to have goals/scope beyond the capabilities of the current team. This is valuable information that can help drive growth decisions.

Reply:
```suggestion
- **Scalable:** Sourcegraph scales to any private codebase and the size of the open-source universe. Users can trust that the result set is drawn from the entire set of code they care about, and if the result isn't found, it doesn't exist.
- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write. We strive for a search experience where the answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
```

Reply:
```suggestion
- **Scalable:** Sourcegraph scales to any private codebase and the size of the open-source universe. Users can trust that the result set is drawn from the entire set of code they care about, and if the result isn't found, it doesn't exist.
- **Expressive:** It is possible to construct a search query to find exactly the results I’m looking for. The tools, syntax, filters, are provided to construct the queries I want to write. We strive for a search experience where the answer to “Can I do ‘X’ with Sourcegraph?” will always be yes.
```